code/game/turfs/simulated.dm

```diff
@@ -55,13 +60,15 @@
 			else
 				wet_overlay = image('icons/effects/water.dmi', src, "wet_static")
 		overlays += wet_overlay
-	if(!infinite)
-		spawn(rand(790, 820)) // Purely so for visual effect
-			if(!istype(src, /turf/simulated)) //Because turfs don't get deleted, they change, adapt, transform, evolve and deform. they are one and they are all.
-				return
-			MakeDry(wet_setting)
+	if(time == INFINITY)
+		return
+	if(time == null)
+		time =	rand(790, 820)
+	addtimer(CALLBACK(src, /turf/simulated./proc/MakeDry, wet_setting), time)
```

Comment:
I am like 99% sure you can just do `.proc/MakeDry` here. I did a quick test of it just to be sure, poured some water on a tile, and it dried just fine.

Reply:
Will change once able!